### ISSUE TITLE
key-spacing

### ISSUE THREAD

**Issue by jamiewinder** [palantir/tslint#991]
Unless I'm missing something, there is no way to enforce whitespace after a property key in an object literal:

``` typescript

    const b = {
        a : 1, // bad
        b: 2 // good
    };
```

I think this is covered in ESLint by [key-spacing](http://eslint.org/docs/rules/key-spacing)

**Comment by adidahiya** [comment]
This would be best as an enhancement to the `whitespace` rule. We should probably update that rule to mirror how `typedef-whitespace` works since #971.

**Comment by micky2be** [comment]
Looking forward

**Comment by johnwiseheart** [comment]
This is pretty old, and a strange feature since we already have a rule disallowing trailing whitespace. Going to close, feel free to re-open/make a new issue if you disagree.

**Comment by adanski** [comment]
@johnwiseheart, how can be the rule requested by @jamiewinder enforced with currently available rules?

**Comment by xemlock** [comment]
@adanski @jamiewinder Looks like this cannot be done by the built-in rules alone. `whitespace.check-type` checks only for space after colon, and `typedef-whitespace`, although more configurable, works only for type definitions.

So I wrote [whitespace-before-colon](https://github.com/xemlock/tslint-whitespace-before-colon) rule which complements those rules, and which checks spaces before colons in object literals and destructuring assignments.
